File: badminton-ai/backend/app/schemas/analytics.py
from __future__ import annotations

from typing import Any

from pydantic import BaseModel, ConfigDict


class ShotEventRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    frame: int
    timestamp: float
    player_id: int | None
    shot_type: str
    confidence: float
    speed_km_h: float
    angle_deg: float
    landing_x: float | None
    landing_y: float | None
    quality_score: int | None
    quality_grade: str | None


class PlayerTrackRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    frame: int
    player_id: int
    team: str
    x: float
    y: float
    speed_ms: float
    confidence: float


class CoachingReportRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    provider: str
    model: str
    prompt_version: str
    report_json: dict[str, Any]
    report_text: str

File: badminton-ai/frontend/src/types/analytics.ts
export interface ShotEventRead {
  id: string;
  frame: number;
  timestamp: number;
  player_id: number | null;
  shot_type: string;
  confidence: number;
  speed_km_h: number;
  angle_deg: number;
  landing_x: number | null;
  landing_y: number | null;
  quality_score: number | null;
  quality_grade: string | null;
}

export interface PlayerTrackRead {
  id: string;
  frame: number;
  player_id: number;
  team: string;
  x: number;
  y: number;
  speed_ms: number;
  confidence: number;
}

export interface CoachingReportRead {
  provider: string;
  model: string;
  prompt_version: string;
  report_json: {
    strengths?: string[];
    weaknesses?: string[];
    tactical_analysis?: string[];
    training_recommendations?: string[];
    summary?: string;
  };
  report_text: string;
}
